### DISCUSSION TITLE
Error in GroupCoverages

### DISCUSSION THREAD

**Discussion by EraslanBas** [GreenleafLab/ArchR#1887] [Questions / Documentation]
Hi,

For some groups I am getting very small number of peaks. When I looked at these groups, I observed that there is something wrong with the names of the group coverage files of these samples (attached, STIM_10nM, STIM_100nM, STIM_10uM).  Would you mind helping me with this issue?

<img width="576" alt="Screen Shot 2023-03-18 at 6 27 16 PM" src="https://user-images.githubusercontent.com/45662603/226148835-17fe9251-e207-4cb5-b366-9cbe36b6700a.png">

Thanks in advance.

**Comment by rcorces** [comment]
You will need to provide more information. In particular a log file from `addGroupCoverages()` would be a good place to start

**Reply by EraslanBas** [reply]
Thank you for your reply Ryan! Attached is the log file of the whole run. Unlike other conditions STIM_10uM does not have _repX file names. 


[ArchR-addGroupCoverages-27da2b6033b2-Date-2023-03-19_Time-18-31-38.log](https://github.com/GreenleafLab/ArchR/files/11021333/ArchR-addGroupCoverages-27da2b6033b2-Date-2023-03-19_Time-18-31-38.log)

**Reply by rcorces** [reply]
I'm not 100% sure whats going on. Looking at your log file hasnt yielded a lot of insight. I would start by trying to identify why ArchR thinks that condition shouldnt get replicates. The easiest way to do this is to use `returnGroups = TRUE` and tweak some of the other input parameters. Minimally you could show me the output of `returnGroups = TRUE` and the number of cells per group (e.g. using `table()`)

**Reply by EraslanBas** [reply]
Thank you Ryan! STIM_10uM and NOSTIM_10uM conditions are the only two conditions where each "Sample" (in the coldata)  has more cells than the  "minCells" parameter of addGroupCoverages(), which is set to 80. (attached are number of cells of the STIM_10uM  and an example condition)

<img width="632" alt="Screen Shot 2023-03-21 at 11 50 30 AM" src="https://user-images.githubusercontent.com/45662603/226712479-ba47e36a-0a2d-4074-ba4c-6f52701d2134.png">


<img width="602" alt="Screen Shot 2023-03-21 at 11 49 59 AM" src="https://user-images.githubusercontent.com/45662603/226712655-abbd8d72-a7ff-4219-a951-2b4305e6cbcb.png">


In the  addGroupCoverages() function I saw the piece of code below, can this be the reason that these two conditions are named differently while generating the group coverages?

<img width="559" alt="Screen Shot 2023-03-21 at 11 56 42 AM" src="https://user-images.githubusercontent.com/45662603/226713054-b5bfbfa9-aae8-4f20-9b38-137c0cde887f.png">

**Reply by rcorces** [reply]
Can you run the below code and send me the 4 RDS files it produces? I think I can help debug with that. You can send to archr.devs@gmail.com

```R
#I based these values on your log file
ArchRProj <- #YOUR ARCHR PROJ HERE
groupBy <- "compoundDosage"
useLabels <- TRUE
sampleLabels <- "Sample"
minCells <- 80
maxCells <- 500
maxFragments <- 2.5e+07
minReplicates <- 2
maxReplicates <- 5
sampleRatio <- 0.8
kmerLength <- 6
threads <- 90
returnGroups <- FALSE
parallelParam <- NULL
force <- TRUE
verbose <- TRUE

#####################################################
#Groups 
#####################################################
cellNames <- rownames(getCellColData(ArchRProj))
groups <- getCellColData(ArchRProj, groupBy, drop = TRUE)
if(any(is.na(groups))){
    cellNames <- cellNames[!is.na(groups)]
    groups <- groups[!is.na(groups)]
}
uniqueGroups <- gtools::mixedsort(unique(groups))
tableGroups <- table(groups)[uniqueGroups]


x <- which(uniqueGroups == "STIM_10uM")

cellColData <- getCellColData(ArchRProj)

saveRDS(object = cellNames, file = "cellNames.RDS")
saveRDS(object = groups, file = "groups.RDS")
saveRDS(object = uniqueGroups, file = "uniqueGroups.RDS")
saveRDS(object = cellColData, file = "cellColData.RDS")
```
